internal/controller/streams.go

```diff
@@ -707,6 +719,11 @@ func ConnectStreams(ctx context.Context, logger *zap.Logger, envMap EnvironmentC
 			sm := &streamManager{
 				streamClient:       streamClient,
 				bufferedGrpcSyncer: bufferedGrpcSyncer,
+				FlowCache: NewFlowCache(
+					20*time.Second, // TODO: Make the active timeout configurable.
+					1000,           // TODO: Make the maxFlows capacity configurable.
+					make(chan Flow, 10),
```

Comment:
I changed my mind.
Please increase the output flow buffer to 100.
That will reduce the blocking for input flows that hit the cache.
```suggestion
					make(chan Flow, 100),
```